_posts/2018-12-06.Rust-1.31.md

```diff
@@ -0,0 +1,338 @@
+---
+layout: post
+title: "Announcing Rust 1.31"
+author: The Rust Core Team
+---
+
+The Rust team is happy to announce a new version of Rust, 1.31.0. Rust is a
+systems programming language focused on safety, speed, and concurrency.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.31.0 is as easy as:
+
+```bash
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.31.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1310-2018-12-06
+
+## What's in 1.31.0 stable
+
+Rust 1.31 may be the most exciting release since Rust 1.0! Included in this release is the
+first iteration of "Rust 2018," but there's more than just that! This is going to be a long
+post, so here's a table of contents:
+
+* [Rust 2018](#rust-2018)
+  * [Non-lexical lifetimes](#non-lexical-lifetimes)
+  * [Module system changes](#module-system-changes)
+* [More lifetime elision rules](#more-lifetime-elision-rules)
+* [`const fn`](#const-fn)
+* [Tool Lints](#tool-lints)
+* [Library stabilizations](#library-stabilizations)
+* [Cargo features](#cargo-features)
+* [Contributors](#contributors-to-1310)
+
+### Rust 2018
+
+We wrote about Rust 2018 [first in
+March](https://blog.rust-lang.org/2018/03/12/roadmap.html), [and then in
+July](https://blog.rust-lang.org/2018/07/27/what-is-rust-2018.html).
+For some more background about the *why* of Rust 2018, please go read those
+posts; there's a lot to cover in the release announcement, and so we're going
+to focus on the *what* here.
+
+Let's create a new project with Cargo:
+
+```console
+$ cargo new foo
+```
+
+Here's the contents of `Cargo.toml`:
+
+```toml
+[package]
+name = "foo"
+version = "0.1.0"
+authors = ["Your Name <you@example.com>"]
+edition = "2018"
+
+[dependencies]
+```
+
+A new key has been added under `[package]`: `edition`. You'll note it's been
+set to `2018`. You can also set it to `2015`, which is the default if the key
+does not exist.
+
+By using Rust 2018, you unlock some new features that are not allowed in Rust 2015.
+And eventually, we'll be turning on some new warnings that help you use these
+new features more effectively, as well.
+
+But it's important to note that each package you use can be in either 2015 or
```

Comment:
```suggestion
It is important to note that each packag can be in either 2015 or
```
The connection to the previous paragraph didn't feel that immediate to have "But ..." imo.